server/services/netatmo/lib/netatmo.loadDeviceDetails.js

```diff
@@ -0,0 +1,139 @@
+const { default: axios } = require('axios');
+const logger = require('../../../utils/logger');
+const { API, SUPPORTED_MODULE_TYPE, SUPPORTED_CATEGORY_TYPE } = require('./utils/netatmo.constants');
+
+/**
+ * @description Discover Netatmo cloud devices.
+ * @param {object} homeData - House.
+ * @returns {Promise} List of discovered devices.
+ * @example
+ * await loadDevicesDetails();
+ */
+async function loadDeviceDetails(homeData) {
+  const { rooms: roomsHomeData, modules: modulesHomeData, id: homeId } = homeData;
+
+  logger.debug('loading devices details in home id: ', homeId, '...');
+  const paramsForm = {
+    home_id: homeId,
+  };
+  try {
+    const responseGetHomestatus = await axios({
+      url: `${API.HOMESTATUS}`,
+      method: 'get',
+      headers: { accept: API.HEADER.ACCEPT, Authorization: `Bearer ${this.accessToken}` },
+      data: paramsForm,
+    });
+    const { body: bodyGetHomestatus, status: statusGetHomestatus } = responseGetHomestatus.data;
+    const { rooms: roomsHomestatus, modules: modulesHomestatus } = bodyGetHomestatus.home;
+    let listDevices = [];
+    if (statusGetHomestatus === 'ok') {
+      let thermostats;
+      let modulesThermostat = [];
+      if (
+        modulesHomeData.find((moduleHomeData) => moduleHomeData.type === SUPPORTED_MODULE_TYPE.THERMOSTAT) !== undefined
+      ) {
+        const deviceThermostats = await this.loadThermostatDetails();
+        thermostats = deviceThermostats.thermostats;
+        modulesThermostat = deviceThermostats.modules;
+      }
+      listDevices = modulesHomestatus
+        ? (
+            await Promise.all(
+              modulesHomestatus.map(async (module) => {
```

Comment:
I don't find any async code in this function? If it's sync, use just .map instead of Promise.all

Reply:
Indeed !!